src/ai/default/aspect_attacks.cpp

```diff
@@ -259,7 +259,10 @@ void aspect_attacks_base::do_attack_analysis(
 				}
 			}
 
-			int best_leadership_bonus = under_leadership(*unit_itor, tiles[j]);
+			int best_leadership_bonus = 0;
+			for(const attack_type& a : unit_itor->attacks()) {
+				best_leadership_bonus= a.under_leadership();
```

Comment:
i dont know what it does but it loks wrong, as it basicially calculated and under_eadership for all of the units attacks, but ignores the result for everythgin but that last attacks

Reply:
ok i will put the old function